
diff -ruN libraries/MarketParamsLib.sol libraries/MarketParamsLib.sol
--- libraries/MarketParamsLib.sol	2023-08-29 09:59:37.937583556 +0200
+++ libraries/MarketParamsLib.sol	2023-08-29 10:16:10.519752188 +0200
@@ -10,8 +10,6 @@
 library MarketParamsLib {
     /// @notice Returns the id of the market `marketParams`.
     function id(MarketParams memory marketParams) internal pure returns (Id marketParamsId) {
-        assembly ("memory-safe") {
-            marketParamsId := keccak256(marketParams, mul(5, 32))
-        }
+        marketParamsId = Id.wrap(keccak256(abi.encode(marketParams)));
     }
 }
